libbeat/common/field_test.go

```diff
@@ -189,3 +191,41 @@ func TestGetKeys(t *testing.T) {
 		assert.Equal(t, test.keys, test.fields.GetKeys())
 	}
 }
+
+func TestFieldUnpack(t *testing.T) {
+	tests := []struct {
+		cfg   MapStr
+		field Field
+		err   bool
+	}{
+		{
+			cfg:   MapStr{"object_type": "scaled_float", "object_type_mapping_type": "float", "scaling_factor": 10},
+			field: Field{ObjectType: "scaled_float", ObjectTypeMappingType: "float", ScalingFactor: 10},
+			err:   false,
+		}, {
+			cfg: MapStr{"object_type_params": []MapStr{
+				{"object_type": "scaled_float", "object_type_mapping_type": "float", "scaling_factor": 100}}},
+			field: Field{ObjectTypeParams: []ObjectTypeCfg{{ObjectType: "scaled_float", ObjectTypeMappingType: "float", ScalingFactor: 100}}},
+			err:   false,
+		}, {
+			cfg: MapStr{
+				"object_type": "scaled_float", "object_type_mapping_type": "float",
+				"object_type_params": []MapStr{{"object_type": "scaled_float", "object_type_mapping_type": "float"}}},
+			err: true,
+		},
+	}
+
+	for _, test := range tests {
+		cfg, err := NewConfigFrom(test.cfg)
+		require.NoError(t, err)
```

Comment:
Each test case could be run in its own subtest so an error here doesn't abort the whole test.